web-registry/src/hooks/useEcocredits.ts

```diff
@@ -23,12 +24,26 @@ const hasBatchBalance = (batchWithBalance: BatchInfoWithBalance): boolean => {
   );
 };
 
+/* If a creditClassId is provided, filter by that. */
```

Comment:
To get the pagination right, I had to apply an optional creditClass filter both during the "initial credits" load (line 78), and after the metadata and CMS content are loaded (line 119).

@victordelval @flagrede Let me know if this sounds right to you.

Reply:
LGTM @mhagel!

Reply:
just a small suggestion for improvement. The predicate function `isOfCreditClass` used to filter, we can curry it and make its use a little more compact and elegant, without the need to pass the items (balances) that are being iterated...

```
const isOfCreditClass =
  (creditClassId?: string) =>
  (balance: BatchBalanceInfo): boolean => {
    if (!creditClassId) return true;
    return balance.batchDenom.startsWith(`${creditClassId}-`);
  };
```
so we can do...
```
...balances.filter(isOfCreditClass(creditClassId))...
```

Reply:
Yeah I like that. Done.